nova/nova.go

```diff
@@ -590,3 +590,38 @@ func (c *Client) SetServerMetadata(serverId string, metadata map[string]string)
 	}
 	return err
 }
+
+// PortFixedIP represents a FixedIP with ip addresses and an associated
+// subnet id.
+type PortFixedIP struct {
+	IPAddress string `json:"ip_address"`
+	SubnetID  string `json:"subnet_id"`
+}
+
+// OSInterface represents an interface attachment to a server.
+type OSInterface struct {
+	FixedIPs   []PortFixedIP `json:"fixed_ips,omitempty"`
+	IPAddress  string        `json:"ip_address"`
+	MacAddress string        `json:"mac_addr,omitempty"`
+	NetID      string        `json:"net_id,omitempty"`
+	PortID     string        `json:"port_id,omitempty"`
+	PortState  string        `json:"port_state,omitempty"`
+}
+
+// ListOSInterfaces lists all the os-interfaces (port interfaces) associated
+// with a given server.
+// https://docs.openstack.org/api-ref/compute/?expanded=list-port-interfaces-detail
+func (c *Client) ListOSInterfaces(serverId string) ([]OSInterface, error) {
```

Comment:
Would it be useful to add call to encapsulate: Show Port Interface Details?  /servers/{server_id}/os-interface/{port_id}.  It provides slightly more data.

Reply:
I did think about doing that, I'll make a follow-up card to look at that in another PR.